external/upstream/gauxc/CMakeLists.txt

```diff
@@ -0,0 +1,64 @@
+if(${ENABLE_gauxc})
+    if((NOT "${gauxc_DIR}" STREQUAL "") AND "${IntegratorXX_DIR}" STREQUAL "")
+       set(IntegratorXX_DIR ${gauxc_DIR}/../IntegratorXX)
+    endif()
+
+    find_package(ExchCXX CONFIG PATHS ${gauxc_DIR}/../ExchCXX) 
+    find_package(gauxc 0.0.1 CONFIG) 
+
+    if(${gauxc_FOUND})
+        #== check if GPU-enabled GauXC is found, if GPU is specified ==#
+        get_property(_gpu TARGET gauxc::gauxc PROPERTY GAUXC_HAS_CUDA)
+        
+        if(${gauxc_ENABLE_GPU} AND (NOT ${_gpu}))
+           message(FATAL_ERROR "gauxc_ENABLE_GPU turned on, but selected GauXC install is not built for GPU support!")
+        elseif((NOT ${gauxc_ENABLE_GPU}) AND ${_gpu})
+           message(FATAL_ERROR "gauxc_ENABLE_GPU turned off, but selected GauXC install is built for GPU support!")
+        endif()
+
+        #== normal print-out and library addition ==#
+        get_property(_loc TARGET gauxc::gauxc PROPERTY IMPORTED_LOCATION_RELWITHDEBINFO)
+        #get_property(_loc TARGET gauxc::gauxc PROPERTY IMPORTED_LOCATION)
+        message(STATUS "${Cyan}Found gauxc ${ColourReset}: ${_loc} (found version ${gauxc_VERSION})")
+        add_library(gauxc_external INTERFACE)  # dummy
+
+        if(${CMAKE_INSIST_FIND_PACKAGE_gauxc})
+            message(VERBOSE "Suitable gauxc located externally as user insists")
+        endif()
+
+    else()
+        if(${CMAKE_INSIST_FIND_PACKAGE_gauxc})
+            message(FATAL_ERROR "Suitable gauxc could not be externally located as user insists")
+        endif()
+
+        message(STATUS "Suitable gauxc could not be located, ${Magenta}Building gauxc${ColourReset} instead.")
+
+        include(ExternalProject)
+        ExternalProject_Add(gauxc_external
+            URL https://github.com/wavefunction91/GauXC/archive/issue_56_58/reduce_shell_pair_mem.tar.gz 
+            UPDATE_COMMAND ""
+            CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${STAGED_INSTALL_PREFIX}
+                   -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+                   -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
+                   -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
+                   -DCMAKE_INSTALL_LIBDIR=${CMAKE_INSTALL_LIBDIR}
+                   -DCMAKE_INSTALL_INCLUDEDIR=${CMAKE_INSTALL_INCLUDEDIR}
+                   -DCMAKE_OSX_DEPLOYMENT_TARGET=${CMAKE_OSX_DEPLOYMENT_TARGET}
+                   -DCMAKE_OSX_SYSROOT=${CMAKE_OSX_SYSROOT}
+                   -DGAUXC_ENABLE_CUDA=${gauxc_ENABLE_GPU}
+                   -DCMAKE_CUDA_ARCHITECTURES=70
+                   -DCMAKE_POSITION_INDEPENDENT_CODE=ON
+                   -DBUILD_SHARED_LIBS=OFF
+                   -DGAUXC_ENABLE_MPI=OFF
+                   -DGAUXC_ENABLE_TESTS=OFF
+                   -DGAUXC_ENABLE_HDF5=OFF
+                   -DGAUXC_ENABLE_MAGMA=OFF
```

Comment:
Eventually this should be made configurable

Reply:
Done and done, at least for HDF5/MAGMA! MPI might not happen for a while; Psi4 and MPI have a rather... storied history together. I'm thinking I can add Cutlass here, as well.

Reply:
One open question on the HDF5 side, however: HDF5-supported GauXC utilizes the HighFive library, which is not natively used by Psi4, nor recognized by the Psi4 build when an external HDF5-supported GauXC is used. Thankfully, this is a very easy fix since HighFive is on Conda. But therein is the question: should there be any references to HighFive on the Psi4 side of the build?

Reply:
The public HDF5 API doesn't depend on HighFive, it should only be required by the build (it's header only). HighFive is used in the tests, but there's no need to build or run those in the Psi4 build system.

However, unless you plan on adding a standard path for Psi4 to generate GauXC benchmark files, there's not a huge need to enable the HDF5 interface, that's pretty much the only reason those interfaces exist.

Reply:
Ah, I misread your initial comment as wanting to make both HDF5 and MAGMA configurable, when it was only MAGMA in reality. All right, I will remove the HDF5 configurability then!